build-system/tasks/helpers.js

```diff
@@ -415,7 +415,10 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
    */
   function remapDependenciesPlugin() {
     const remaps = Object.entries(options.remapDependencies).map(
-      ([path, value]) => ({regex: new RegExp(`^${path}$`), value})
+      ([path, value]) => ({
+        regex: new RegExp(`^${path}(\.js|\.jsx|\.ts|\.tsx)?$`),
```

Comment:
Supernit:
```suggestion
        regex: new RegExp(`^${path}(?:\.[jt]sx?)?$`),
```
(See https://regexper.com/#%5E%7Bpath%7D%28%3F%3A%5C.%5Bjt%5Dsx%3F%29%3F%24)

Reply:
I love that regex.

But I think Alan has a point that if we make it ugly it'll be more discoverable when the next dev has to add one more extension or some other random thing.

Reply:
Oh lol, didn't notice Alan's comment before :D
Fair enough, it's far from being a dealbreaker one way or the other